src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2782,6 +2782,176 @@ export default class EthereumApi implements Api {
+   *
+   * @returns The fee history for the returned block range. This can be a subsection of the requested range if not all blocks are available. Includes the oldestBlock, baseFeePerGas, gasUsedRatio, and percentile rewards.
+   * @example for: blockCount: 2, newestBlock: 0x10, rewardPercentiles: [10, 25, 50]
+   * {
+   *   oldestBlock: "0x10",
+   *   reward:[["0x59682f00","0x2a7b749c11","0x2a7b749c11"]],
+   *   baseFeePerGas:["0x40c9459e7","0x420068cc1","0x420068cc1"]
+   *   gasUsedRatio: [0.5750394293729167, 0.50]
+   * };
+   *
+   */
+  @assertArgLength(2, 3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles?: number[]
+  ): Promise<Ethereum.FeeHistoryResult> {
+    let reward = [];
+    const baseFeePerGas = [];
+    const gasUsedRatio = [];
+    const blockchain = this.#blockchain;
+    const PAD_PRECISION = 16; // Matching geth/infura decimal precision formatting
+    const PRECISION = 10 ** PAD_PRECISION;
+    const PRECISION_BIG_INT = Quantity.from(PRECISION).toBigInt();
+    const PRECISION_BIG_INT_PERCENTILE = PRECISION_BIG_INT * 100n;
+    const ZERO_BIG_INT = Quantity.from(0).toBigInt();
+
+    let blocksToFetch = Quantity.toNumber(blockCount);
+    let newestBlockNumber = blockchain.blocks
+      .getEffectiveNumber(newestBlock)
+      .toNumber();
+    let oldestBlock;
+
+    let currentBlockNumber = newestBlockNumber;
+
+    // Because tags are supported in newestBlock, starting from the oldest block
+    // Would require hitting the db for the latest block twice. Once to calc to oldest
+    // and once to process the block.
+    while (blocksToFetch > 0 && currentBlockNumber >= 0) {
+      const currentBlock = await blockchain.blocks.get(currentBlockNumber);
+      oldestBlock = Quantity.from(currentBlockNumber).toString();
+
+      const gasUsed = currentBlock.header.gasUsed.toBigInt();
+      const gasLimit = currentBlock.header.gasLimit.toBigInt();
+      const baseFee = currentBlock.header.baseFeePerGas
+        ? currentBlock.header.baseFeePerGas.toBigInt()
+        : ZERO_BIG_INT;
+
+      // gasUsedRatio
+      if (gasUsed === ZERO_BIG_INT) {
+        gasUsedRatio.unshift(0);
+      } else {
+        gasUsedRatio.unshift(
+          Number(
+            `0.${((gasUsed * PRECISION_BIG_INT) / gasLimit)
+              .toString()
+              .padStart(PAD_PRECISION, "0")}`
+          )
+        );
+      }
+
+      // baseFeePerGas
+      if (!baseFee) {
+        baseFeePerGas.unshift("0x0");
+      } else {
+        // Next block fee is based on existing blocks
+        if (currentBlockNumber === newestBlockNumber) {
+          baseFeePerGas.unshift(
+            Quantity.from(Block.calcNextBaseFee(currentBlock)).toString()
+          );
+        }
+        baseFeePerGas.unshift(Quantity.from(baseFee).toString());
```

Comment:
Same as the previous comment
```suggestion
        baseFeePerGas.unshift(Quantity.from(baseFee));
```